src/io/flutter/vmService/VMServiceManager.java

```diff
@@ -530,7 +531,9 @@ public boolean hasServiceExtensionNow(String name) {
   }
 
   public void hasServiceExtension(String name, Consumer<Boolean> onData, Disposable parentDisposable) {
-    Disposer.register(parentDisposable, hasServiceExtension(name, onData));
+    if (!Disposer.isDisposed(parentDisposable)) {
```

Comment:
Do you have any details on what is further up the stack? It seems like something else has gone badly wrong if we are calling this method when parentDisposable has already been disposed.

Reply:
From looking through the source, I think this can only happen if we're still setting up parts of the UI when the app terminates.